bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/AbstractComponent.java

```diff
@@ -154,28 +164,47 @@ public void addChannelTypes(MqttChannelTypeProvider channelTypeProvider) {
      */
     public void removeChannelTypes(MqttChannelTypeProvider channelTypeProvider) {
         channels.values().forEach(v -> v.removeChannelTypes(channelTypeProvider));
-        channelTypeProvider.removeChannelGroupType(getGroupTypeUID());
+        ChannelGroupTypeUID groupTypeUID = channelGroupTypeUID;
+        if (groupTypeUID != null) {
+            channelTypeProvider.removeChannelGroupType(groupTypeUID);
+        }
+    }
+
+    public ChannelUID buildChannelUID(String channelID) {
+        final ChannelGroupUID groupUID = channelGroupUID;
+        if (groupUID != null) {
+            return new ChannelUID(groupUID, channelID);
+        }
+        return new ChannelUID(componentConfiguration.getThingUID(), channelID);
     }
 
     /**
      * Each HomeAssistant component corresponds to a Channel Group Type.
      */
-    public ChannelGroupTypeUID getGroupTypeUID() {
+    public @Nullable ChannelGroupTypeUID getGroupTypeUID() {
         return channelGroupTypeUID;
     }
 
     /**
      * The unique id of this component.
      */
-    public ChannelGroupUID getGroupUID() {
+    public @Nullable ChannelGroupUID getGroupUID() {
         return channelGroupUID;
     }
 
     /**
      * Component (Channel Group) name.
      */
     public String getName() {
-        return channelConfiguration.getName();
+        String result = channelConfiguration.getName();
+
+        if (result == null && channelConfiguration.getDevice() != null) {
+            result = channelConfiguration.getDevice().getName();
```

Comment:
Perhaps a local variable could be introduced to hold the value of `channelConfiguration.getDevice()` so that it would guaranteed that same value is being checked and used.